tools/toolchains/mbed_toolchain.py

```diff
@@ -1093,6 +1095,28 @@ def check_and_add_minimal_printf(self, target):
         ):
             self.flags["common"].append("-DMBED_MINIMAL_PRINTF")
 
+    def check_default_lib_supported(self, target, toolchain):
+        """Check and raise an exception if default_lib not supported for this target."""
+        if  hasattr(target, "default_lib"):
+            if  target.default_lib == "std":
+                if (
+                    hasattr(target, "supported_c_libs") == False
+                    or toolchain not in target.supported_c_libs
+                    or "std" not in target.supported_c_libs[toolchain]
+                ):
+                    raise NotSupportedException(
+                      UNSUPPORTED_OPTION_STD_EXECPTION_STRING
+                    )
+            elif target.default_lib == "small":
+                if (
```

Comment:
The code is repeated. You only the following:
```
if hasattr(target, "default_lib"):
    if (
        hasattr(target, "supported_c_libs") == False
        or toolchain not in target.supported_c_libs
        or target.std not in target.supported_c_libs[toolchain]
    ):
        raise NotSupportedException(
            UNSUPPORTED_OPTION_STD_EXECPTION_STRING.format(target.std)
        )
```
An added benefit is that it also handles unrecognized values.